### PR TITLE
feat: ACN image

### PR DESCRIPTION
**Pull request by Karrenbelt**
We 

This introduces the run_acn_node_standalone script and Dockerfile previously developed by 8ball030. The Dockerfile has been modified:
- to target this repository
- to use git instead of subversion
- updated golang to 1.17.7

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+ENV PATH="${PATH}:/usr/local/go/bin"
+
+# deployment scripts deps: needed only if configuration checks are enabled
+RUN python3 -m pip install pymultihash ecdsa base58
+
+WORKDIR /acn/
+
+# get node source code
+RUN git clone https://github.com/valory-xyz/open-acn /acn/node
```
Comment (DavidMinarsch) [inline Dockerfile:26]: since you're running this locally from the same repo, why not just COPY?

File: run_acn_node_standalone.py [hunk 2]
```diff
@@ -0,0 +1,389 @@
+# -*- coding: utf-8 -*-
```
Comment (DavidMinarsch) [inline run_acn_node_standalone.py:1]: Please explain where this file was taken from and what the diff is (exact links)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+# utils
+RUN apt install -y wget git
+
+# golang
+RUN wget https://dl.google.com/go/go1.17.7.linux-amd64.tar.gz && \
+  tar -xzvf go1.17.7.linux-amd64.tar.gz -C /usr/local && \
+  export PATH=$PATH:/usr/local/go/bin && echo 'export PATH=$PATH:/usr/local/go/bin' >> ~/.bashrc && \
+  mkdir $HOME/go
+
+ENV PATH="${PATH}:/usr/local/go/bin"
```
Comment (0xArdi) [inline Dockerfile:18]: Why not start from a golang image?

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+WORKDIR /acn/
+
+# get node source code
+RUN git clone https://github.com/valory-xyz/open-acn /acn/node
+
+# get deployment script
+COPY run_acn_node_standalone.py /acn/
+
+# build node
+RUN cd /acn/node && go build
```
Comment (0xArdi) [inline Dockerfile:32]: We can do this in a separate `acn-build` stage. We don't need golang compile time dependencies on the final image.

File: run_acn_node_standalone.py (hunk 2 above)
Comment (Karrenbelt) [reply run_acn_node_standalone.py:1]: [original Dockerfile](https://github.com/valory-xyz/open-aea/blob/main/scripts/acn/Dockerfile)

diff: 
```
# utils
-RUN apt install -y wget
+RUN apt install -y wget subversion
 
 # golang
-RUN wget https://dl.google.com/go/go1.17.7.linux-amd64.tar.gz && \
-  tar -xzvf go1.17.7.linux-amd64.tar.gz -C /usr/local && \
+RUN wget https://dl.google.com/go/go1.13.8.linux-amd64.tar.gz && \
+  tar -xzvf go1.13.8.linux-amd64.tar.gz -C /usr/local && \
   export PATH=$PATH:/usr/local/go/bin && echo 'export PATH=$PATH:/usr/local/go/bin' >> ~/.bashrc && \
   mkdir $HOME/go
 
@@ -23,7 +23,7 @@ RUN python3 -m pip install pymultihash ecdsa base58
 WORKDIR /acn/
 
 # get node source code
-COPY . /acn/node
+RUN svn export https://github.com/valory-xyz/open-aea/trunk/packages/open_aea/connections/p2p_libp2p/libp2p_node/ /acn/node
```

File: README.md [hunk 5]
```diff
@@ -78,3 +78,43 @@ The ACN has the following protocols:
 - unregister (dealt with by DHT defaults)
 - DHT default protocols in libp2p
 - message delivery protocol
+
+## Dockerfile
+
+We need a private key, we can be created using open-aea:
```
Comment (DavidMinarsch) [inline README.md:84]: Please make clear here that this key is for testing purposes only.

File: Dockerfile (hunk 3 above)
Comment (Karrenbelt) [reply Dockerfile:18]: we could slim the image down, not in this PR though

File: README.md [hunk 6]
```diff
@@ -78,3 +78,70 @@ The ACN has the following protocols:
...
+cp bootstrap_peer/cosmos_private_key.txt ./node
+```
+
+The ACN node can then be deployed as follows:
+```fish
+docker run -it -p 11000:11000 -p 9000:9000 -v (pwd)/node:/node valory/acn-node:v0.1.0 --key-file /node/cosmos_private_key.txt --uri 0.0.0.0:9000 --uri-external 0.0.0.0:10000 --uri-delegate 0.0.0.0:11000
+```
+
+The expected output should look as follows:
+```fish
```
Comment (8ball030) [inline README.md:123]: change from fish to bash please, itswill throw other people otherwise

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,38 @@
+FROM ubuntu:20.04
+
+USER root
```
Comment (8ball030) [inline Dockerfile:3]: Why are we using the root user as default?
Is there a reason behind this?

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+# deployment scripts deps: needed only if configuration checks are enabled
+RUN python3 -m pip install pymultihash ecdsa base58
+
+WORKDIR /acn/
+
+# get node source code
+COPY . /acn/node
+
+# get deployment script
+COPY run_acn_node_standalone.py /acn/
```
Comment (8ball030) [inline Dockerfile:29]: This suggests your structure is incorrect,

you shouldnt need to do 2 sets of copy.

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,38 @@
+FROM ubuntu:20.04
+
```
Comment (8ball030) [inline Dockerfile:2]: Better practise would be to inheirt from one of the python containers, then you wouldnt need to install python and you get much smaller images,

File: Dockerfile [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+USER root
+
+RUN apt-get update && apt-get upgrade -y
+
+RUN apt install -y python3 python3-pip
+
+# utils
+RUN apt install -y wget
```
Comment (8ball030) [inline Dockerfile:10]: These apt-gets should be combined

File: Dockerfile (hunk 7 above)
Comment (Karrenbelt) [reply Dockerfile:3]: not that I know of, copied from [original Dockerfile](https://github.com/valory-xyz/open-aea/blob/main/scripts/acn/Dockerfile)

File: README.md [hunk 11]
```diff
@@ -96,31 +96,32 @@ cd bootstrap_peer && \
...
 The private key file can be copied to the `node/` folder and mounted to the images:
-```fish
+```bash
 cp bootstrap_peer/cosmos_private_key.txt ./node
 ```
 
 The ACN node can then be deployed as follows:
-```fish
+```bash
 docker run -it -p 11000:11000 -p 9000:9000 -v (pwd)/node:/node valory/acn-node:v0.1.0 --key-file /node/cosmos_private_key.txt --uri 0.0.0.0:9000 --uri-external 0.0.0.0:10000 --uri-delegate 0.0.0.0:11000
```
Comment (8ball030) [inline README.md:120]: This command wont run in bash

**Review by 8ball030 — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**